manifest/docker_schema2_list.go

```diff
@@ -101,11 +101,25 @@ func (list *Schema2List) ChooseInstance(ctx *types.SystemContext) (digest.Digest
 		wantedOS = ctx.OSChoice
 	}
 
+	var fallback digest.Digest
+
 	for _, d := range list.Manifests {
 		if d.Platform.Architecture == wantedArch && d.Platform.OS == wantedOS {
-			return d.Digest, nil
+			// TODO It should be possible to somehow use runtime.GOARM to construct a default VariantChoice
+			if ctx.VariantChoice == "" {
```

Comment:
Why not
if ctx == "" || d.Platform.Variant == ctx.VariantChoice {
 				return d.Digest, nil
}

Reply:
That slipped, I'll fix it